src/org/abymaly/code/magnets/Magnets.java

```diff
@@ -2,4 +2,158 @@
 
 public class Magnets {
 
-}
+    public Signs[][] board;
+    private final int size;
+    public int possibleMoves = 0;
+
+    public Magnets(int size) {
+        this.size = size;
+        this.board = new Signs[size][size];
+    }
+
+    public Magnets() {
+        this.size = 4;
+    }
+
+    class StartPosition {
+        int x;
+        int y;
+
+        public StartPosition() {
+        }
+    }
+
+    StartPosition startPosition = new StartPosition();
+
+    public void printBoard(Signs[][] board) {
+        String newLine = "\n";
+        StringBuffer sbLine = new StringBuffer();
+
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                if (board[i][j].equals(Signs.EMPTY)) {
+                    sbLine.append(" _ ");
+                } else {
+                    sbLine.append(" ");
+                    sbLine.append(board[i][j]);
+                    sbLine.append(" ");
+                }
+            }
+            sbLine.append(newLine);
+        }
+        System.out.println(sbLine);
+    }
+
+    public void setStartPosition(Signs[][] board, int x, int y) {
+        board[x][y] = Signs.O;
+    }
+
+    public void setData(Signs[][] board, int x, int y, Signs value) {
+        board[x][y] = value;
+    }
+
+    public Signs[][] emptyBoard () {
+        Signs [][] newBoard = new Signs[size][size];
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                setData(newBoard, i, j, Signs.EMPTY);
+            }
+        }
+        return newBoard;
+    }
+
+    public void refreshStartPosition(Signs[][] board) {
+        setStartPosition(board, startPosition.x, startPosition.y);
+    }
+
+    public void emptyStartPosition (Signs[][] board) {
+        setData(board, startPosition.x, startPosition.y, Signs.EMPTY);
+    }
+
+    public void getStartPosition(Signs[][] board) {
+
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                if (board[i][j].equals(Signs.O)) {
+                    startPosition.x = i;
+                    startPosition.y = j;
+                }
+            }
+        }
+    }
+
+    public boolean moveTop(Signs[][] board) {
+        if (startPosition.x == 0 || board[(startPosition.x) - 1][startPosition.y].equals(Signs.X)) {
```

Comment:
Ten sam komentarz by był do każdej z czterech funkcji
Jest ok, ale chodzi o czytelność.
Ten warunek jest długi, trudno się go czyta. W takich sytuacjach jest ok rozbić go na dwa oddzielne, po każdym `return`. A po returnach można pominąć `else`, wtedy czytający widzi - to były sytuacje szczególne (`return false`), a jeśli nie zaszły do jest mięsko ruch.

To też nie jest błąd: enum to jest w Javie (w C++ zresztą też) taka twarda liczba jak na przykład `17`, to nie jest obiekt. `equals` jest zbędny (bo to nie są referencje), wystarczy `==`.